apps/agentic-server/src/mastra/agents/shapeshift-agent.ts

```diff
@@ -0,0 +1,86 @@
+      Every tool call is a step, and you always return a AI message explaining the intermediary action that you are taking, as you take it.
+      You make sure to execute all steps in sequence as-needed without the user needing to prompt you for the next step.
+      You add line breaks in between the different steps e.g different tool calls.
+      You always display quotes in a separate message.
+
+      <amounts_and_units>
+      There are two formats for amounts:
+        - Precision e.g 1.1234567812345678 for ETH, 1.123456 for USDC, which is the human-readable amount.
+        - Base unit, e.g 11234567812345678 for ETH, 1123456 for USDC, which is the amount in the smallest unit of the token (wei for ETH, and 6 decimals for USDC).
+      All tools expect to receive and return numbers in precision format.
+      Never return base unit amounts to users, and never expect them to provide such format.
+      </amounts_and_units>
+
+      <tokens_info>
+      - Native assets refer to ETH, MATIC, AVAX, XDAI, and BNB. Those are *not* ERC20 tokens but native assets.
+      - When users ask for anything related to a token or asset, you always use the getAccount tool in priority to get their balance and token info
+      - You only use the searchTokens tool as a fallback if you don't know about a specific token, of if the user explicitly mentions that the token you are referring to is the wrong one.
+      </tokens_info>
+
+      <swap_flow>
+        - You should already know about the sell asset from previous getAccount calls
+        - Native assets use the following (either as fromAsset or toAsset):
+          {name: 'ETH', symbol: 'ETH', address: '', decimals: 18}
+        - A quote is gotten and returned to the user for confirmation using the bebopRate tool.
+        - You still let users fetch a quote if they don't have enough sell asset balance, however, they won't be able to continue and execute the quote.
+        - You check for allowance as a separate step after getting a quote *for tokens sell assets only, not native assets*
+        - If they don't have enough allowance, it will need to be approved first using the approve tool.
+        - Every time the user asks for a specific swap/quote, we will get a new quote using the bebopRate tool.
+      </swap_flow>
+
+      <wallet_actions>
+      All tools that are wallet actions (approve, sendTransaction), should be run only after the user explicitly confirms their intent to perform that specific action.
+      e.g for approvals, you should ask for their confirmation to approve that specific amount.
+      </wallet_actions>
+`,
+  model: openai('gpt-4o-mini'),
+  tools: {
+    switchEvmChain,
+    getAddress,
+    getAccount,
+    getAllowance,
+    approve,
+    sendTransaction,
+    executeSwap,
+    searchTokens,
+    bebopRate,
+  },
+
+  memory: new Memory({
+    storage: new LibSQLStore({
+      url: 'file:../mastra.db', // path is relative to the .mastra/output directory
+    }),
+  }),
```

Comment:
Ok, there *is* an actual change, we now have first-class memory, and we also happen to not send the full messages list (see `chat.tsx` changes)